src/pool/srv_pool.c

```diff
@@ -967,11 +968,20 @@ ds_pool_svc_stop(uuid_t pool_uuid)
 	d_iov_set(&id, pool_uuid, sizeof(uuid_t));
 
 	rc = ds_rsvc_stop(DS_RSVC_CLASS_POOL, &id, RDB_NIL_TERM, false /* destroy */);
-	if (rc == -DER_ALREADY) {
-		D_DEBUG(DB_MD, DF_UUID": ds_rsvc_stop: "DF_RC"\n", DP_UUID(pool_uuid), DP_RC(rc));
-		rc = 0;
+	if (rc != 0) {
+		if (rc == -DER_ALREADY) {
+			D_DEBUG(DB_MD, DF_UUID": ds_rsvc_stop: "DF_RC"\n",
+				DP_UUID(pool_uuid), DP_RC(rc));
+			rc = 0;
+		}
+		D_GOTO(out, rc);
+	}
+
+	if (bio_nvme_configured(SMD_DEV_TYPE_META)) {
+		rc = vos_pool_kill(pool_uuid, VOS_POF_SMALL | VOS_POF_EXCL | VOS_POF_RDB);
```

Comment:
I'm afraid I have a few concerns:

- If no strong reason, we should avoid including vos.h in the current file.
- Adding the destroy here raises the question: Why not call `ds_rsvc_stop` with `true /* destroy */`? We should try fixing the latter to take care of NVMe by somehow calling `rdb_destroy`.
- Calling `bio_nvme_configured` seems unnecessary, but I'm not sure.

Could this PR wait a bit for me to discuss the concerns with @NiuYawei, please?

Reply:
No concern from my side to wait after @NiuYawei .

For the third point, I put it to not change the behavior with NVDIM and non persistant RAM-FS system configuration.

For the second point, I have to make some investigations... From what I remember I tried to call `ds_rsvc_stop()` with `true /* destroy */`, and the blob were not deleted.  However, I will check this point as I am not sure of it.

Reply:
@liw and I walked through the RDB and regular pool destroy code yesterday, our conclusion is that we'd handle the RDB blob deletion in the common pool destroy code path, otherwise, the RDB blob could still be leaked in certain scenarios.

Reply:
Thanks for helping, Niu.

@knard-intel, the outcome of the discussion Niu and I had is to call `vos_pool_kill` in `ds_mgmt_hdlr_tgt_destroy` (the common pool destroy code path) and `cleanup_leftover_cb` (the engine recovery code path).

One aspect we didn't discuss is that we still need to fix `ds_rsvc_stop(..., true /* destroy */)` to delete the blob(s) properly, probably by replacing the `remove` call in `stop` with `vos_pool_kill`.

If you prefer, I could make the changes by pushing more commits to this PR. Please let me know how you like to proceed when you are back. Thanks.